### PR TITLE
working CI

### PR DESCRIPTION
**Pull request by LennardBoediger**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+stages:
+  - pull-github
+  - build
+  - deploy
+
+variables:
+  GIT_STRATEGY: clone			#This seems to have no effect also set in webinterface
+  GIT_DEPTH: 0					#This seems to have no effect also set in webinterface
```
Comment (szszszsz) [inline .gitlab-ci.yml:8]: I am not sure `0` here is a correct value. The smallest one is `1` AFAIK. Can you confirm?
Perhaps this is why it does not work.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  image: ubuntu:20.04
+  tags:
+    - lxc
+  stage: build
+  before_script:
+    - apt update -y
+    - apt install -y git
+    - apt install -y make
+    - apt install -y gcc
+    - apt install -y python3
```
Comment (szszszsz) [inline .gitlab-ci.yml:77]: These can be installed in one command.

File: docker_build.mk [hunk 3]
```diff
@@ -15,7 +15,6 @@ uncached-docker-build-toolchain:
 docker-build-all:
 	docker run --rm -v "$(CURDIR)/builds:/builds" \
 					-v "$(CURDIR):/solo-base:ro" \
-					-u $(shell id -u ${USER}):$(shell id -g ${USER}) \
```
Comment (szszszsz) [inline docker_build.mk:18]: This one is handy for a local run so the resulted files were written with running user privileges, although it was not working lately. Can you fix that? If not, please leave it as a commented line.

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    - git submodule update --init --recursive
```
Comment (szszszsz) [inline .gitlab-ci.yml:78]: Would be nice to have `--recommend-shallow` here as well. Maybe for later - can you add TODO note?

**Review by szszszsz — CHANGES_REQUESTED** [review]
Looks good! Quick final changes and we could go.

File: .gitlab-ci.yml (hunk 1 above)
Comment (LennardBoediger) [reply .gitlab-ci.yml:8]: 0 means max depth we need that for git describe

File: docker_build.mk (hunk 3 above)
Comment (LennardBoediger) [reply docker_build.mk:18]: done

File: .gitlab-ci.yml (hunk 4 above)
Comment (LennardBoediger) [reply .gitlab-ci.yml:78]: done

File: .gitlab-ci.yml (hunk 2 above)
Comment (LennardBoediger) [reply .gitlab-ci.yml:77]: done

**Review by szszszsz — APPROVED** [review]

**Pull request merged**